src/main/java/frc/robot/subsystems/BatterySubsystem.java

```diff
@@ -0,0 +1,199 @@
+    // Adds sim voltage reader tab
+    sbSimVoltage = btTab.add("Simulation Voltage", 0).withSize(2, 1).withPosition(0, 0).getEntry();
+    // Adds general timer reader tab
+    sbTimer = btTab.add("Timer", 0).withSize(2, 1).withPosition(0, 2).getEntry();
+    // Adds high current timer reader tab
+    sbTimerHighCurrent =
+        btTab.add("High Current Timer", 0).withSize(2, 1).withPosition(0, 1).getEntry();
+    // Adds timer change reader tab
+    sbTimerChange = btTab.add("Change Timer", 0).withSize(0, 1).withPosition(4, 0).getEntry();
+    // Starts general timer
+    timer.start();
+    DriverStationSim.setSendError(true);
+  }
+
+  public void periodic() {
+    // Sets shuffleboard tabs to their respective values
+    sbVoltage.setDouble(getVoltage());
+    sbInputCurrent.setDouble(getInputCurrent());
+    sbSimVoltage.setNumber(powerDistribution.getVoltage());
+    sbTimer.setDouble(getGeneralTimer());
+    sbTimerHighCurrent.setDouble(getHighCurrentTimer());
+    sbTimerChange.setBoolean(checkTimer()); // Replace checkVoltage() with checkTimer() if necessary
+    checkCurrent();
+    checkVoltage();
+  }
+
+  public double getVoltage() {
+    return powerDistribution.getVoltage();
+  }
+
+  public double getInputCurrent() {
+    if (Robot.isSimulation()) {
+      return powerDistribution.getCurrent(1);
+    } else {
+      return powerDistribution.getTotalCurrent();
+    }
+  }
+
+  // If the current is below a certain level, stops the high current timer
+  public void checkCurrent() {
+    if (getInputCurrent() < Constants.highBatteryCurrentThreshold) {
+      stopHCTimer();
+    } else {
+      startHCTimer();
+    }
+  }
+
+  // Displays alerts based on the state of the timer
+  public boolean checkTimer() {
+    if (currentTimer.hasElapsed(Constants.timeInSecondsHighCurrentRed)) {
+      DriverStation.reportWarning("Change the Battery Now! (HCTR)", false);
+      return true;
+    } else if (timer.hasElapsed(Constants.timeInSecondsGeneralRed)) {
+      DriverStation.reportWarning("Change the Battery Now! (GTR)", false);
+      return true;
+    } else if (currentTimer.hasElapsed(Constants.timeInSecondsHighCurrentYellow)) {
+      DriverStation.reportWarning("Change the Battery Soon! (HCTY)", false);
+      return false;
+    } else if (timer.hasElapsed(Constants.timeInSecondsGeneralYellow)) {
+      DriverStation.reportWarning("Change the Battery Soon! (GTY)", false);
+      return false;
+    }
+    return false;
+  }
+
+  public double getGeneralTimer() {
+    Preferences.setDouble("Battery General Timer", timer.get() + storedGeneralTime);
+    return timer.get() + storedGeneralTime;
+  }
+
+  public double getHighCurrentTimer() {
+    Preferences.setDouble("Battery High Current Timer", currentTimer.get() + storedHighCurrentTime);
+    return currentTimer.get() + storedHighCurrentTime;
+  }
+
+  private void startHCTimer() {
+    currentTimer.start();
+  }
+
+  private void stopHCTimer() {
+    currentTimer.stop();
+  }
```

Comment:
Methods usually only need to be created to add a capability/access to the object from an outside source (public methods), or to reuse code that is commonly used and relatively long. Since this case is neither, I would rather just use currentTimer.start and stop directly where we needed it instead of having methods for them.

Reply:
done, methods have been removed